### PR TITLE
Fix check_infer_symbolic_pass

### PR DESCRIPTION
**Pull request by JiaWenxuan**
### PR types

Bug fixes

### PR changes

Others

### Description

pcard-76996
fix check_infer_symbolic_pass to support Tensorlist for value.

### CODE REVIEW

File: paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc [hunk 1]
```diff
@@ -23,23 +23,29 @@
...
 namespace {
 
-std::string SprintShape(const std::vector<std::int64_t>& shape) {
-  std::string str = "[";
-  for (std::int64_t value : shape) {
-    str += std::to_string(value);
-    if (value != shape.back()) {
+std::string SprintShape(const std::vector<std::vector<std::int64_t>>& shapes) {
+  std::string str;
+  for (std::vector<std::int64_t> shape : shapes) {
```
Comment (jiahy0825) [inline paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc]:
```suggestion
  for (const auto& shape : shapes) {
```

File: paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc [hunk 2]
```diff
@@ -23,23 +23,29 @@
...
+    str += "[";
+    for (std::int64_t value : shape) {
+      str += std::to_string(value);
+      if (value != shape.back()) {
```
Comment (jiahy0825) [inline paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc]: shape 里可能有重复的值，这里会出问题

File: paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc [hunk 3]
```diff
@@ -51,23 +57,55 @@ void PrintProgram(pir::ModuleOp m, const std::string& mgs) {
...
+    return dynamic_shapes;
   }
+  symbol::ShapeOrDataDimExprs shape_or_data =
+      shape_analysis.GetShapeOrDataForValue(value);
+  auto lambdas = symbol::Overloaded{
+      [&](const symbol::TensorShapeOrDataDimExprs& tensor_shape_or_data) {
+        dynamic_shapes.push_back(GetShapeFromTensor(tensor_shape_or_data));
+      },
+      [&](const symbol::TensorListShapeOrDataDimExprs& tensor_list) {
+        for (symbol::TensorShapeOrDataDimExprs tensor_shape_or_data :
```
Comment (jiahy0825) [inline paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc]: const auto&
省掉不必要的拷贝

File: paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc (hunk 1 above)
Comment (JiaWenxuan) [reply paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc]: 已修改

File: paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc (hunk 3 above)
Comment (JiaWenxuan) [reply paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc]: 已修改

File: paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc (hunk 2 above)
Comment (JiaWenxuan) [reply paddle/cinn/hlir/dialect/operator/transforms/group_merge/check_infer_symbolic_pass.cc]: 已修改

**Review by jiahy0825 — APPROVED** [review]
LGTM

**Pull request merged**